src/keys.rs

```diff
@@ -35,7 +35,9 @@ impl SpendingKey {
         let sk = SpendingKey(sk);
         // If ask = 0, discard this key.
```

Comment:
```suggestion
        // If ask = 0, discard this key. We call `derive_inner` rather than
        // `SpendAuthorizingKey::from` here because we only need to know
        // whether ask = 0; the adjustment to potentially negate ask is not needed.
```

Reply:
This comment is not quite correct here: `SpendAuthorizingKey::from` panics if `ask = 0` because it relies on the invariant that a `SpendingKey` cannot be constructed such that `ask = 0`. So we _cannot_ call it here (and this is exactly why `dreive_inner` exists).

The additional comment about `ask` negation is maybe still useful, but if we did need it, it would be factored out of `SpendAuthorizingKey::from`.

Reply:
I adjusted the suggestion.